2024-03-22 00:25:44,663 [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x10a2ad430>
  on_reload: <function OnReload.on_reload at 0x10a2ad550>
  when_ready: <function WhenReady.when_ready at 0x10a2ad670>
  pre_fork: <function Prefork.pre_fork at 0x10a2ad790>
  post_fork: <function Postfork.post_fork at 0x10a2ad8b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x10a2ad9d0>
  worker_int: <function WorkerInt.worker_int at 0x10a2adaf0>
  worker_abort: <function WorkerAbort.worker_abort at 0x10a2adc10>
  pre_exec: <function PreExec.pre_exec at 0x10a2add30>
  pre_request: <function PreRequest.pre_request at 0x10a2ade50>
  post_request: <function PostRequest.post_request at 0x10a2adee0>
  child_exit: <function ChildExit.child_exit at 0x10a2c7040>
  worker_exit: <function WorkerExit.worker_exit at 0x10a2c7160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10a2c7280>
  on_exit: <function OnExit.on_exit at 0x10a2c73a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10a2c74c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2024-03-22 00:25:44 -0400] [72124] [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x10a2ad430>
  on_reload: <function OnReload.on_reload at 0x10a2ad550>
  when_ready: <function WhenReady.when_ready at 0x10a2ad670>
  pre_fork: <function Prefork.pre_fork at 0x10a2ad790>
  post_fork: <function Postfork.post_fork at 0x10a2ad8b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x10a2ad9d0>
  worker_int: <function WorkerInt.worker_int at 0x10a2adaf0>
  worker_abort: <function WorkerAbort.worker_abort at 0x10a2adc10>
  pre_exec: <function PreExec.pre_exec at 0x10a2add30>
  pre_request: <function PreRequest.pre_request at 0x10a2ade50>
  post_request: <function PostRequest.post_request at 0x10a2adee0>
  child_exit: <function ChildExit.child_exit at 0x10a2c7040>
  worker_exit: <function WorkerExit.worker_exit at 0x10a2c7160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10a2c7280>
  on_exit: <function OnExit.on_exit at 0x10a2c73a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10a2c74c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
2024-03-22 00:25:44,663 [INFO] Starting gunicorn 21.2.0
[2024-03-22 00:25:44 -0400] [72124] [INFO] Starting gunicorn 21.2.0
2024-03-22 00:25:44,663 [DEBUG] Arbiter booted
[2024-03-22 00:25:44 -0400] [72124] [DEBUG] Arbiter booted
2024-03-22 00:25:44,664 [INFO] Listening at: http://127.0.0.1:8001 (72124)
[2024-03-22 00:25:44 -0400] [72124] [INFO] Listening at: http://127.0.0.1:8001 (72124)
2024-03-22 00:25:44,664 [INFO] Using worker: sync
[2024-03-22 00:25:44 -0400] [72124] [INFO] Using worker: sync
2024-03-22 00:25:44,667 [INFO] Booting worker with pid: 72125
[2024-03-22 00:25:44 -0400] [72125] [INFO] Booting worker with pid: 72125
2024-03-22 00:25:44,687 [DEBUG] 1 workers
[2024-03-22 00:25:44 -0400] [72124] [DEBUG] 1 workers
2024-03-22 00:25:46,722 [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
[2024-03-22 00:25:46 -0400] [72125] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
2024-03-22 00:25:46,725 [INFO] Worker exiting (pid: 72125)
[2024-03-22 00:25:46 -0400] [72125] [INFO] Worker exiting (pid: 72125)
2024-03-22 00:25:46,849 [ERROR] Worker (pid:72125) exited with code 3
[2024-03-22 00:25:46 -0400] [72124] [ERROR] Worker (pid:72125) exited with code 3
2024-03-22 00:25:46,850 [ERROR] Shutting down: Master
[2024-03-22 00:25:46 -0400] [72124] [ERROR] Shutting down: Master
2024-03-22 00:25:46,850 [ERROR] Reason: Worker failed to boot.
[2024-03-22 00:25:46 -0400] [72124] [ERROR] Reason: Worker failed to boot.
2024-03-22 00:27:23,300 [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x102071430>
  on_reload: <function OnReload.on_reload at 0x102071550>
  when_ready: <function WhenReady.when_ready at 0x102071670>
  pre_fork: <function Prefork.pre_fork at 0x102071790>
  post_fork: <function Postfork.post_fork at 0x1020718b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1020719d0>
  worker_int: <function WorkerInt.worker_int at 0x102071af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x102071c10>
  pre_exec: <function PreExec.pre_exec at 0x102071d30>
  pre_request: <function PreRequest.pre_request at 0x102071e50>
  post_request: <function PostRequest.post_request at 0x102071ee0>
  child_exit: <function ChildExit.child_exit at 0x10208b040>
  worker_exit: <function WorkerExit.worker_exit at 0x10208b160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10208b280>
  on_exit: <function OnExit.on_exit at 0x10208b3a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10208b4c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2024-03-22 00:27:23 -0400] [72485] [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x102071430>
  on_reload: <function OnReload.on_reload at 0x102071550>
  when_ready: <function WhenReady.when_ready at 0x102071670>
  pre_fork: <function Prefork.pre_fork at 0x102071790>
  post_fork: <function Postfork.post_fork at 0x1020718b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1020719d0>
  worker_int: <function WorkerInt.worker_int at 0x102071af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x102071c10>
  pre_exec: <function PreExec.pre_exec at 0x102071d30>
  pre_request: <function PreRequest.pre_request at 0x102071e50>
  post_request: <function PostRequest.post_request at 0x102071ee0>
  child_exit: <function ChildExit.child_exit at 0x10208b040>
  worker_exit: <function WorkerExit.worker_exit at 0x10208b160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10208b280>
  on_exit: <function OnExit.on_exit at 0x10208b3a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10208b4c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
2024-03-22 00:27:23,301 [INFO] Starting gunicorn 21.2.0
[2024-03-22 00:27:23 -0400] [72485] [INFO] Starting gunicorn 21.2.0
2024-03-22 00:27:23,302 [DEBUG] Arbiter booted
[2024-03-22 00:27:23 -0400] [72485] [DEBUG] Arbiter booted
2024-03-22 00:27:23,302 [INFO] Listening at: http://127.0.0.1:8001 (72485)
[2024-03-22 00:27:23 -0400] [72485] [INFO] Listening at: http://127.0.0.1:8001 (72485)
2024-03-22 00:27:23,302 [INFO] Using worker: sync
[2024-03-22 00:27:23 -0400] [72485] [INFO] Using worker: sync
2024-03-22 00:27:23,306 [INFO] Booting worker with pid: 72491
[2024-03-22 00:27:23 -0400] [72491] [INFO] Booting worker with pid: 72491
2024-03-22 00:27:23,339 [DEBUG] 1 workers
[2024-03-22 00:27:23 -0400] [72485] [DEBUG] 1 workers
2024-03-22 00:27:24,993 [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
[2024-03-22 00:27:24 -0400] [72491] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
2024-03-22 00:27:25,000 [INFO] Worker exiting (pid: 72491)
[2024-03-22 00:27:25 -0400] [72491] [INFO] Worker exiting (pid: 72491)
2024-03-22 00:27:25,162 [ERROR] Worker (pid:72491) exited with code 3
[2024-03-22 00:27:25 -0400] [72485] [ERROR] Worker (pid:72491) exited with code 3
2024-03-22 00:27:25,163 [ERROR] Shutting down: Master
[2024-03-22 00:27:25 -0400] [72485] [ERROR] Shutting down: Master
2024-03-22 00:27:25,164 [ERROR] Reason: Worker failed to boot.
[2024-03-22 00:27:25 -0400] [72485] [ERROR] Reason: Worker failed to boot.
2024-03-22 00:28:52,356 [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x109f28430>
  on_reload: <function OnReload.on_reload at 0x109f28550>
  when_ready: <function WhenReady.when_ready at 0x109f28670>
  pre_fork: <function Prefork.pre_fork at 0x109f28790>
  post_fork: <function Postfork.post_fork at 0x109f288b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x109f289d0>
  worker_int: <function WorkerInt.worker_int at 0x109f28af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x109f28c10>
  pre_exec: <function PreExec.pre_exec at 0x109f28d30>
  pre_request: <function PreRequest.pre_request at 0x109f28e50>
  post_request: <function PostRequest.post_request at 0x109f28ee0>
  child_exit: <function ChildExit.child_exit at 0x109f42040>
  worker_exit: <function WorkerExit.worker_exit at 0x109f42160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x109f42280>
  on_exit: <function OnExit.on_exit at 0x109f423a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x109f424c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2024-03-22 00:28:52 -0400] [72783] [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x109f28430>
  on_reload: <function OnReload.on_reload at 0x109f28550>
  when_ready: <function WhenReady.when_ready at 0x109f28670>
  pre_fork: <function Prefork.pre_fork at 0x109f28790>
  post_fork: <function Postfork.post_fork at 0x109f288b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x109f289d0>
  worker_int: <function WorkerInt.worker_int at 0x109f28af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x109f28c10>
  pre_exec: <function PreExec.pre_exec at 0x109f28d30>
  pre_request: <function PreRequest.pre_request at 0x109f28e50>
  post_request: <function PostRequest.post_request at 0x109f28ee0>
  child_exit: <function ChildExit.child_exit at 0x109f42040>
  worker_exit: <function WorkerExit.worker_exit at 0x109f42160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x109f42280>
  on_exit: <function OnExit.on_exit at 0x109f423a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x109f424c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
2024-03-22 00:28:52,356 [INFO] Starting gunicorn 21.2.0
[2024-03-22 00:28:52 -0400] [72783] [INFO] Starting gunicorn 21.2.0
2024-03-22 00:28:52,357 [DEBUG] Arbiter booted
[2024-03-22 00:28:52 -0400] [72783] [DEBUG] Arbiter booted
2024-03-22 00:28:52,357 [INFO] Listening at: http://127.0.0.1:8001 (72783)
[2024-03-22 00:28:52 -0400] [72783] [INFO] Listening at: http://127.0.0.1:8001 (72783)
2024-03-22 00:28:52,357 [INFO] Using worker: sync
[2024-03-22 00:28:52 -0400] [72783] [INFO] Using worker: sync
2024-03-22 00:28:52,361 [INFO] Booting worker with pid: 72784
[2024-03-22 00:28:52 -0400] [72784] [INFO] Booting worker with pid: 72784
2024-03-22 00:28:52,402 [DEBUG] 1 workers
[2024-03-22 00:28:52 -0400] [72783] [DEBUG] 1 workers
2024-03-22 00:28:53,922 [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
[2024-03-22 00:28:53 -0400] [72784] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 3, in <module>
    from indexing import *
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 317, in <module>
    class Indexer:
  File "/Users/luyan/Code/MagicPrompt/indexing.py", line 325, in Indexer
    max_docs: int = -1, doc_augment_dict: dict[int, list[str]] | None = None) -> InvertedIndex:
TypeError: unsupported operand type(s) for |: 'types.GenericAlias' and 'NoneType'
2024-03-22 00:28:53,928 [INFO] Worker exiting (pid: 72784)
[2024-03-22 00:28:53 -0400] [72784] [INFO] Worker exiting (pid: 72784)
2024-03-22 00:28:54,061 [ERROR] Worker (pid:72784) exited with code 3
[2024-03-22 00:28:54 -0400] [72783] [ERROR] Worker (pid:72784) exited with code 3
2024-03-22 00:28:54,062 [ERROR] Shutting down: Master
[2024-03-22 00:28:54 -0400] [72783] [ERROR] Shutting down: Master
2024-03-22 00:28:54,062 [ERROR] Reason: Worker failed to boot.
[2024-03-22 00:28:54 -0400] [72783] [ERROR] Reason: Worker failed to boot.
2024-03-22 00:33:38,099 [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x103045430>
  on_reload: <function OnReload.on_reload at 0x103045550>
  when_ready: <function WhenReady.when_ready at 0x103045670>
  pre_fork: <function Prefork.pre_fork at 0x103045790>
  post_fork: <function Postfork.post_fork at 0x1030458b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1030459d0>
  worker_int: <function WorkerInt.worker_int at 0x103045af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x103045c10>
  pre_exec: <function PreExec.pre_exec at 0x103045d30>
  pre_request: <function PreRequest.pre_request at 0x103045e50>
  post_request: <function PostRequest.post_request at 0x103045ee0>
  child_exit: <function ChildExit.child_exit at 0x10305f040>
  worker_exit: <function WorkerExit.worker_exit at 0x10305f160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10305f280>
  on_exit: <function OnExit.on_exit at 0x10305f3a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10305f4c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2024-03-22 00:33:38 -0400] [73760] [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x103045430>
  on_reload: <function OnReload.on_reload at 0x103045550>
  when_ready: <function WhenReady.when_ready at 0x103045670>
  pre_fork: <function Prefork.pre_fork at 0x103045790>
  post_fork: <function Postfork.post_fork at 0x1030458b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1030459d0>
  worker_int: <function WorkerInt.worker_int at 0x103045af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x103045c10>
  pre_exec: <function PreExec.pre_exec at 0x103045d30>
  pre_request: <function PreRequest.pre_request at 0x103045e50>
  post_request: <function PostRequest.post_request at 0x103045ee0>
  child_exit: <function ChildExit.child_exit at 0x10305f040>
  worker_exit: <function WorkerExit.worker_exit at 0x10305f160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x10305f280>
  on_exit: <function OnExit.on_exit at 0x10305f3a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x10305f4c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
2024-03-22 00:33:38,099 [INFO] Starting gunicorn 21.2.0
[2024-03-22 00:33:38 -0400] [73760] [INFO] Starting gunicorn 21.2.0
2024-03-22 00:33:38,100 [DEBUG] Arbiter booted
[2024-03-22 00:33:38 -0400] [73760] [DEBUG] Arbiter booted
2024-03-22 00:33:38,100 [INFO] Listening at: http://127.0.0.1:8001 (73760)
[2024-03-22 00:33:38 -0400] [73760] [INFO] Listening at: http://127.0.0.1:8001 (73760)
2024-03-22 00:33:38,100 [INFO] Using worker: sync
[2024-03-22 00:33:38 -0400] [73760] [INFO] Using worker: sync
2024-03-22 00:33:38,104 [INFO] Booting worker with pid: 73764
[2024-03-22 00:33:38 -0400] [73764] [INFO] Booting worker with pid: 73764
2024-03-22 00:33:38,177 [DEBUG] 1 workers
[2024-03-22 00:33:38 -0400] [73760] [DEBUG] 1 workers
2024-03-22 00:33:40,648 [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 12, in <module>
    from gevent import pywsgi
ModuleNotFoundError: No module named 'gevent'
[2024-03-22 00:33:40 -0400] [73764] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.9/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luyan/Code/MagicPrompt/app.py", line 12, in <module>
    from gevent import pywsgi
ModuleNotFoundError: No module named 'gevent'
2024-03-22 00:33:40,654 [INFO] Worker exiting (pid: 73764)
[2024-03-22 00:33:40 -0400] [73764] [INFO] Worker exiting (pid: 73764)
2024-03-22 00:33:40,859 [ERROR] Worker (pid:73764) exited with code 3
[2024-03-22 00:33:40 -0400] [73760] [ERROR] Worker (pid:73764) exited with code 3
2024-03-22 00:33:40,860 [ERROR] Shutting down: Master
[2024-03-22 00:33:40 -0400] [73760] [ERROR] Shutting down: Master
2024-03-22 00:33:40,860 [ERROR] Reason: Worker failed to boot.
[2024-03-22 00:33:40 -0400] [73760] [ERROR] Reason: Worker failed to boot.
2024-03-22 00:34:12,873 [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x105929430>
  on_reload: <function OnReload.on_reload at 0x105929550>
  when_ready: <function WhenReady.when_ready at 0x105929670>
  pre_fork: <function Prefork.pre_fork at 0x105929790>
  post_fork: <function Postfork.post_fork at 0x1059298b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1059299d0>
  worker_int: <function WorkerInt.worker_int at 0x105929af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x105929c10>
  pre_exec: <function PreExec.pre_exec at 0x105929d30>
  pre_request: <function PreRequest.pre_request at 0x105929e50>
  post_request: <function PostRequest.post_request at 0x105929ee0>
  child_exit: <function ChildExit.child_exit at 0x105944040>
  worker_exit: <function WorkerExit.worker_exit at 0x105944160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x105944280>
  on_exit: <function OnExit.on_exit at 0x1059443a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x1059444c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2024-03-22 00:34:12 -0400] [73973] [DEBUG] Current configuration:
  config: config.py
  wsgi_app: None
  bind: ['127.0.0.1:8001']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /Users/luyan/Code/MagicPrompt
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 501
  group: 20
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: dev_gunicorn_access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: dev_gunicorn_error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  logconfig_json: None
  syslog_addr: unix:///var/run/syslog
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: app:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x105929430>
  on_reload: <function OnReload.on_reload at 0x105929550>
  when_ready: <function WhenReady.when_ready at 0x105929670>
  pre_fork: <function Prefork.pre_fork at 0x105929790>
  post_fork: <function Postfork.post_fork at 0x1059298b0>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x1059299d0>
  worker_int: <function WorkerInt.worker_int at 0x105929af0>
  worker_abort: <function WorkerAbort.worker_abort at 0x105929c10>
  pre_exec: <function PreExec.pre_exec at 0x105929d30>
  pre_request: <function PreRequest.pre_request at 0x105929e50>
  post_request: <function PostRequest.post_request at 0x105929ee0>
  child_exit: <function ChildExit.child_exit at 0x105944040>
  worker_exit: <function WorkerExit.worker_exit at 0x105944160>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x105944280>
  on_exit: <function OnExit.on_exit at 0x1059443a0>
  ssl_context: <function NewSSLContext.ssl_context at 0x1059444c0>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
2024-03-22 00:34:12,873 [INFO] Starting gunicorn 21.2.0
[2024-03-22 00:34:12 -0400] [73973] [INFO] Starting gunicorn 21.2.0
2024-03-22 00:34:12,874 [DEBUG] Arbiter booted
[2024-03-22 00:34:12 -0400] [73973] [DEBUG] Arbiter booted
2024-03-22 00:34:12,874 [INFO] Listening at: http://127.0.0.1:8001 (73973)
[2024-03-22 00:34:12 -0400] [73973] [INFO] Listening at: http://127.0.0.1:8001 (73973)
2024-03-22 00:34:12,874 [INFO] Using worker: sync
[2024-03-22 00:34:12 -0400] [73973] [INFO] Using worker: sync
2024-03-22 00:34:12,878 [INFO] Booting worker with pid: 73974
[2024-03-22 00:34:12 -0400] [73974] [INFO] Booting worker with pid: 73974
2024-03-22 00:34:12,918 [DEBUG] 1 workers
[2024-03-22 00:34:12 -0400] [73973] [DEBUG] 1 workers
2024-03-22 00:34:21,776 [DEBUG] GET /
[2024-03-22 00:34:21 -0400] [73974] [DEBUG] GET /
2024-03-22 00:34:23,514 [DEBUG] Ignoring connection reset
[2024-03-22 00:34:23 -0400] [73974] [DEBUG] Ignoring connection reset
2024-03-22 00:34:23,515 [DEBUG] GET /
[2024-03-22 00:34:23 -0400] [73974] [DEBUG] GET /
2024-03-22 00:34:25,079 [DEBUG] GET /static/css/style_search.css
[2024-03-22 00:34:25 -0400] [73974] [DEBUG] GET /static/css/style_search.css
2024-03-22 00:34:25,091 [DEBUG] GET /static/css/style_index.css
[2024-03-22 00:34:25 -0400] [73974] [DEBUG] GET /static/css/style_index.css
2024-03-22 00:34:26,128 [DEBUG] GET /static/src/magnifying-glass-solid.svg
[2024-03-22 00:34:26 -0400] [73974] [DEBUG] GET /static/src/magnifying-glass-solid.svg
2024-03-22 00:34:26,142 [DEBUG] GET /static/src/image-regular.svg
[2024-03-22 00:34:26 -0400] [73974] [DEBUG] GET /static/src/image-regular.svg
2024-03-22 00:34:26,149 [DEBUG] GET /static/src/feather-solid.svg
[2024-03-22 00:34:26 -0400] [73974] [DEBUG] GET /static/src/feather-solid.svg
2024-03-22 00:34:26,239 [DEBUG] GET /static/js/script.js
[2024-03-22 00:34:26 -0400] [73974] [DEBUG] GET /static/js/script.js
2024-03-22 00:34:30,278 [DEBUG] GET /static/src/wand-magic-sparkles-solid.svg
[2024-03-22 00:34:30 -0400] [73974] [DEBUG] GET /static/src/wand-magic-sparkles-solid.svg
2024-03-22 00:34:31,790 [DEBUG] POST /search
[2024-03-22 00:34:31 -0400] [73974] [DEBUG] POST /search
2024-03-22 00:34:32,156 [INFO] Handling signal: int
[2024-03-22 00:34:32 -0400] [73973] [INFO] Handling signal: int
2024-03-22 00:34:32,267 [INFO] Worker exiting (pid: 73974)
[2024-03-22 00:34:32 -0400] [73974] [INFO] Worker exiting (pid: 73974)
2024-03-22 00:34:33,245 [ERROR] Worker (pid:73974) was sent SIGINT!
[2024-03-22 00:34:33 -0400] [73973] [ERROR] Worker (pid:73974) was sent SIGINT!
2024-03-22 00:34:33,285 [INFO] Shutting down: Master
[2024-03-22 00:34:33 -0400] [73973] [INFO] Shutting down: Master
